packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx

```diff
@@ -18,6 +18,23 @@ describe('<DateField /> - Editing', () => {
 
   const { clickOnInput } = buildFieldInteractions({ clock });
 
+  const selectSection = (input: HTMLInputElement, activeSectionIndex: number) => {
```

Comment:
I'm trying another approach than `cursorPosition` or `valueToSelect`.

If at some point we add a way to get the section array, then this would probably be a lot easier.
For now it support all the formats with `/` or `:` which should be the vast majority of the scenarios.